js/components/Debugger.tsx

```diff
@@ -57,10 +66,38 @@ export const Debugger = ({ apple2 }: DebuggerProps) => {
         debug?.step();
     }, [debug]);
 
-    const doMemoryPage = useCallback((event: JSX.TargetedMouseEvent<HTMLInputElement>) => {
+    const doLoadAddress = useCallback((event: JSX.TargetedEvent<HTMLInputElement>) => {
+        setLoadAddress(event.currentTarget.value);
+    }, []);
+    const doRunCheck = useCallback((event: JSX.TargetedEvent<HTMLInputElement>) => {
+        setRun(event.currentTarget.checked);
+    }, []);
+
+    const doMemoryPage = useCallback((event: JSX.TargetedEvent<HTMLInputElement>) => {
         setMemoryPage(event.currentTarget.value);
     }, []);
 
+    const doChooseFile = useCallback((handles: FileSystemFileHandle[]) => {
+        if (debug && handles.length === 1) {
+            spawn(async () => {
+                const file = await handles[0].getFile();
+                let atAddress = parseInt(loadAddress, 16) || 0x800;
```

Comment:
Should this validation be done in the `doMemoryPage` callback? As it is, the user never sees the actual address that's used if it doesn't validate. Or, if you don't want to validate there since it's set on every keypress, maybe call `setMemoryPage(toHex(loadAddress, 4))` or similar?